### DISCUSSION TITLE
Starter packs and relationship to underlying moderation lists

### DISCUSSION THREAD

**Discussion by ruyimarone** [bluesky-social/atproto#3099] [Clarifications and "Why?"]
Hi, my apologies if this isn't the best place to ask this question. I wasn't sure what falls under the boundary of the atproto protocol vs the bluesky app interface. 

While playing with the python atproto api, I found that starter packs (as with [this api](https://atproto.blue/en/latest/atproto/atproto_client.models.app.bsky.graph.get_starter_pack.html#module-atproto_client.models.app.bsky.graph.get_starter_pack)) have a corresponding list given by `response.starter_pack.list.uri` (e.g. `at://did:plc:.../app.bsky.graph.list/...` ). 

I can view that list in my browser by constructing a url e.g. `https://bsky.app/profile/marcmarone.com/lists/<list_uri_from_above>`. 

I can also follow example code like here: https://github.com/MarshalX/atproto/blob/main/examples/advanced_usage/add_user_to_list.py , to add a user to the underlying starter pack list. When I do this, I see the user also appears in the starter pack interface. 

## My questions:
1. When viewed in my browser, the list underlying the starter pack shows as a "moderation list". Can I treat that list as the cannonical list backing the starterpack? Meaning I can make programmatic modifications to the list and see the changes reflected in the starter pack that I've shared with others. 

2. I notice that I don't have an option to pin this list to home, as I do for normal lists. Is this because of its special status underlying the starter pack? Or perhaps because it's called a "moderation" list? 

3. There seem to be limits on the size of starter packs. What happens to my starter pack if I grow the underlying list? 

My goals are to have a starter pack of users and corresponding list, that crucially can be viewed as a feed (seems pin to home is the best mechanism for this). My plan was to use a script to sync a starter pack (or several) to a list. But if there's a better way to go about this, please share, thanks!

**Comment by KevinCathcart** [comment]
What you are looking for here is to have a starter pack that is referencing a list of type `app.bsky.graph.defs#curatelist`, which are the ones that can be pinned as a feed. Normally starter packs reference a list of type `app.bsky.graph.defs#referencelist`, which indicates a list meant to be referenced from elsewhere, not used directly, so it won't show up under lists or moderation lists. I'd be surprised if the starter pack code really cared about the underlying list being a referencelist. Indeed, I did a quick test, and it looked like it was perfectly happy with a curatelist.

The easy way to do this, is to create a pinnable curation list like normal, and then hand craft the app.bsky.graph.starterpack entry:
```json
{
    "collection": "app.bsky.graph.starterpack",
    "repo": "did:plc:[MY_PLC_ID]",
    "record": {
        "name": "[STARTER_PACK_NAME]",
        "list": "at://did:plc:[MY_PLC_ID]/app.bsky.graph.list/[EXISTING_LIST_IDENTIFIER]",
        "feeds": [],
        "createdAt": "[TIMESTAMP]",
        "$type": "app.bsky.graph.starterpack"
    }
}
```
Obviously I filled in the text in square brackets with appropriate values.  Once you create this record, you should be able to simply maintain the existing list as normal, and the starter pack would remain in sync.

It *might* be possible to update an existing starter pack's list to change the type, or to update an existing starter pack to swap out the list, but I've not tried this. Some form of caching might prevent it, much like how post edits do not show up in the appview.

As for how I created the record, I did it in the browser console with code like the following, which I based on what was sent when I used to UI to create a list, thus some of the headers might be overkill.
```js
fetch("https://[MY_PDS].[MY_PDS_REGION].host.bsky.network/xrpc/com.atproto.repo.createRecord", {
  "headers": {
    "accept": "*/*",
    "accept-language": "en-US,en;q=0.9",
    "atproto-accept-labelers": "did:plc:ar7c4by46qjdydhdevvrndac;redact",
    "authorization": "Bearer [MY_TOKEN]",
    "cache-control": "no-cache",
    "content-type": "application/json",
    "pragma": "no-cache",
    "priority": "u=1, i",
    "sec-fetch-dest": "empty",
    "sec-fetch-mode": "cors",
    "sec-fetch-site": "cross-site"
  },
  "referrer": "https://bsky.app/",
  "referrerPolicy": "origin-when-cross-origin",
  "body": "{\"collection\":\"app.bsky.graph.starterpack\",\"repo\":\"did:plc:[MY_PLC_ID]\",\"record\":{\"name\":\"[STARTER_PACK_NAME]\",\"list\":\"at://did:plc:[MY_PLC_ID]/app.bsky.graph.list/[EXISTING_LIST_IDENTIFIER]\",\"feeds\":[],\"createdAt\":\"[TIMESTAMP]\",\"$type\":\"app.bsky.graph.starterpack\"}}",
  "method": "POST",
  "mode": "cors"
});
```
